### ISSUE TITLE
removal of encode_json makes functionality impossible to reach

### ISSUE THREAD

**Issue by RunOrVeith** [lidatong/dataclasses-json#229]
Please consider the following example:

```python
from dataclasses import dataclass
from enum import Enum

from dataclasses_json import dataclass_json


class Dummy(Enum):
    A = "a"
    B = "b"


@dataclass_json()
@dataclass()
class EnumResponse:
    which: Dummy


obj = EnumResponse(which=Dummy.A)

print(obj.to_dict()["which"])



```

The issue is in [this commit](https://github.com/lidatong/dataclasses-json/commit/1887d996192048e2721a4b9587824e7f548b2475)
This now prints `Dummy.A`. Before removal of the parameter, we got `a`, if we used `obj.to_dict(encode_json=True)`.
This is now impossible to get (unless you do `to_json`, and then `json.loads`, which is not very nice imho).

It makes sense to remove passing the argument when inside `to_json`, because the json library does not require it any more.
However, I would like to argue to keep the parameter exposed inside `to_dict`.
The use case here is that we use the dataclasses to build payloads for an API that takes loaded json content (external library).
Now if there is an enum inside, we can no longer use the dataclass.to_dict method to create the payload, since we need to manually case enum values.

**TLDR**: I  propose to re-expose the `encode_json` parameter inside `to_dict` so that enums can get get converted (and probably things like sets etc. as well).

**Comment by eqperes** [comment]
Same problem here. I also used it to encode enums.

**Comment by RunOrVeith** [comment]
@lidatong I can undo that change mentioned above in a pull request. It's keeping us from updating the library, and it seems a few other people as well.
What do you think? Did you have a special reason for removing it?

**Comment by lidatong** [comment]
oh, i didn't realize this parameter had this effect... thought it was an internal API parameter that was accidentally exposed... should've commented / prevents this naming / functionality combo in the relevant PR. thanks for offering the fix, i have just released a version reverting it in 0.5.2

i'm sorry for the lengthy delay -- i have been taking a break for a while. didn't mean for this to be a blocker

**Comment by RunOrVeith** [comment]
Great, thank you! I think we can close this issue and #240 in that case
